src/browser/components/TabNavigation/Navigation.tsx

```diff
@@ -156,48 +159,47 @@ class Navigation extends Component<NavigationProps, NavigationState> {
         )
       })
 
-    const getContentToShow = (openDrawer?: null | string) => {
-      if (openDrawer) {
+    const getContentToShow = (selectedDrawerName?: null | string) => {
+      if (selectedDrawerName) {
         const filteredList = topNavItems
           .concat(bottomNavItems)
-          .filter(item => item.name.toLowerCase() === openDrawer)
+          .filter(item => item.name.toLowerCase() === selectedDrawerName)
         const TabContent = filteredList[0].content
         return <TabContent />
       }
       return null
     }
-    const topNavItemsList = buildNavList(topNavItems, this.state.drawerContent)
+    const topNavItemsList = buildNavList(
+      topNavItems,
+      this.props.selectedDrawerName
+    )
     const bottomNavItemsList = buildNavList(
       bottomNavItems,
-      this.state.drawerContent
+      this.props.selectedDrawerName
     )
 
+    const isDrawerVisible = () => this.state.transitionState !== Closed
+
     const drawerWidth =
-      this.props.openDrawer === GUIDE_DRAWER_ID
+      this.props.selectedDrawerName === GUIDE_DRAWER_ID
         ? LARGE_DRAWER_WIDTH
         : STANDARD_DRAWER_WIDTH
     const useFullWidth =
       this.state.transitionState === Open ||
       this.state.transitionState === Opening
     const width = useFullWidth ? drawerWidth : 0
+
     return (
       <StyledSidebar>
         <StyledTabsWrapper>
           <StyledTopNav>{topNavItemsList}</StyledTopNav>
           <StyledBottomNav>{bottomNavItemsList}</StyledBottomNav>
         </StyledTabsWrapper>
-        <StyledDrawer
-          width={width}
-          ref={ref => {
-            if (ref) {
-              // Remove old listeners so we don't get multiple callbacks.
-              // This function is called more than once with same html element
-              ref.removeEventListener('transitionend', this._onTransitionEnd)
-              ref.addEventListener('transitionend', this._onTransitionEnd)
-            }
-          }}
-        >
-          {getContentToShow(this.state.drawerContent)}
+        <StyledDrawer width={width} onTransitionEnd={this.onTransitionEnd}>
+          {isDrawerVisible() &&
+            getContentToShow(
+              this.props.selectedDrawerName || this.state.lastSelectedDrawerName
```

Comment:
Not sure how, but it'd be nice if we could explain why lastSelectedDrawerName is needed. Perhaps a comment? Or some other name? Thoughts?

Reply:
yes I was thinking of naming it something with closing, like closingDrawerName, since it is only used when a drawer is closing to keep the content until width is 0 and it is totally invisible. WDYT?

Reply:
It's somewhat long but calling it `drawerToShowWhileClosing` could do the trick?

Reply:
I'll leave you to pick the name